src/Model/ModelManager.php

```diff
@@ -250,25 +250,23 @@ public function delete($object)
             $entityManager = $this->getEntityManager($object);
             $entityManager->remove($object);
             $entityManager->flush();
-        } catch (\PDOException $e) {
-            throw new ModelManagerException(
-                sprintf('Failed to delete object: %s', ClassUtils::getClass($object)),
-                $e->getCode(),
-                $e
-            );
-        } catch (DBALException $e) {
+        } catch (\Exception $exception) {
             throw new ModelManagerException(
                 sprintf('Failed to delete object: %s', ClassUtils::getClass($object)),
-                $e->getCode(),
-                $e
+                $exception->getCode(),
+                $exception
             );
         }
     }
 
     public function getLockVersion($object)
     {
-        // NEXT_MAJOR: Remove `sonata_deprecation_mute`.
-        $metadata = $this->getMetadata(ClassUtils::getClass($object), 'sonata_deprecation_mute');
+        try {
+            // NEXT_MAJOR: Remove `sonata_deprecation_mute`.
+            $metadata = $this->getMetadata(ClassUtils::getClass($object), 'sonata_deprecation_mute');
+        } catch (\Exception $exception) {
+            return null;
```

Comment:
Catching an exception without logging is dangerous!

Reply:
Please take a look at the explanation https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1243#issue-547793912
I would be happy to get your opinion about the different point.

For this case, theoretically if there is no entityManager/metadata, the object has no lockVersion.
To me, other solution would be,
- if `getMetadata` throw a `ModelManagerException`, I catch it, then I throw instead a `LockException` seems this method is in the LockInterface.
- if `getMetadata` throw only `\InvalidArgumentException`, I don't catch it.
So we should first decide about the listed points.